kpet/data.py

```diff
@@ -387,26 +388,105 @@ def validate_host_type_regex(self):
                                   host_type_regex.pattern,
                                   host_types)
 
+    def convert_tree_arches(self):
+        """
+        Get a list of available arches for each tree
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+
+        trees = {}
+        # Create a dict {tree: arches,} where arches is a list of regexes
+        for tree in self.trees:
+            trees[tree] = []
```

Comment:
You don't need to keep the other tree's arches for each iteration, so just use a single variable. Also, switch to sets to simplify the code and make it shorter. E.g.:

    tree_arches = set()